core/hoc/CompartmentMapping.hoc

```diff
@@ -163,10 +166,10 @@ proc registerMapping () { local cellIndex, activegid, nSoma, nAxon, nDend, nApic
     }
 
     if( has_lfp_manager != 0 ) {
-        lfp_factors = cellDistributor._lfp_manager.read_lfp_factors(activegid, secvec)
+        lfp_factors = cellDistributor._lfp_manager.read_lfp_factors(activegid, indexSec, nAxon, pop_info)
```

Comment:
Wouldnt be easier to get all lfp factors at once and then just pick as many we need in each step?